blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractPreaggregatedIO.java

```diff
@@ -0,0 +1,333 @@
+            put(Granularity.FULL, insertToMetricsPreaggrFullStatement);
+            put(Granularity.MIN_5, insertToMetricsPreaggr5MStatement);
+            put(Granularity.MIN_20, insertToMetricsPreaggr20MStatement);
+            put(Granularity.MIN_60, insertToMetricsPreaggr60MStatement);
+            put(Granularity.MIN_240, insertToMetricsPreaggr240MStatement);
+            put(Granularity.MIN_1440, insertToMetricsPreaggr1440MStatement);
+        }};
+    }
+
+    /**
+     * Asynchronously insert a rolled up metric to the appropriate column family
+     * for a particular granularity
+     *
+     * @param locator
+     * @param collectionTime
+     * @param rollup
+     * @param granularity
+     * @return
+     */
+    public ResultSetFuture putAsync(Locator locator, long collectionTime, Rollup rollup, Granularity granularity, int ttl) {
+
+        Session session = DatastaxIO.getSession();
+
+        // we use batch statement here in case sub classes
+        // override the addRollupToBatch() and provide
+        // multiple statements
+        BatchStatement batch = new BatchStatement();
+        addRollupToBatch(batch, locator, rollup, collectionTime, granularity, ttl);
+        return session.executeAsync(batch);
+    }
+
+    /**
+     * Fetch rollup objects for a {@link com.rackspacecloud.blueflood.types.Locator}
+     * from the specified column family and range.
+     *
+     * @param locator
+     * @param columnFamily
+     * @param range
+     * @return
+     */
+    protected <T extends Rollup> Table<Locator, Long, T> getRollupsForLocator(final Locator locator, String columnFamily, Range range) {
+        return getRollupsForLocators(new ArrayList<Locator>() {{ add(locator); }}, columnFamily, range);
+    }
+
+    /**
+     * Fetch rollup objects for a list of {@link com.rackspacecloud.blueflood.types.Locator}
+     * from the specified column family and range.
+     *
+     * This is a base behavior for most rollup types. IO subclasses can override
+     * this behavior as they see fit.
+     *
+     * @param locators
+     * @param columnFamily
+     * @param range
+     * @return
+     */
+    protected <T extends Rollup> Table<Locator, Long, T> getRollupsForLocators(final List<Locator> locators, String columnFamily, Range range) {
+
+        Table<Locator, Long, T> locatorTimestampRollup = HashBasedTable.create();
+
+        Map<Locator, List<ResultSetFuture>> resultSetFuturesMap = selectForLocatorListAndRange(columnFamily, locators, range);
+
+        for (Map.Entry<Locator, List<ResultSetFuture>> entry : resultSetFuturesMap.entrySet() ) {
+            Locator locator = entry.getKey();
+            try {
+                List<ResultSetFuture> futures = entry.getValue();
+
+                Table<Locator, Long, T> result = toLocatorTimestampRollup(futures, locator, CassandraModel.getGranularity(columnFamily));
+                locatorTimestampRollup.putAll(result);
```

Comment:
the ```toLocatorTimestampRollup()``` is where the future.getInterruptibly() get called. Do you think I should move the try-catch inside the toLocatorTimestampRollup(), which currently doesn't do any Instrumentation calls? Maybe I could just declare the exceptions as stated in the Datastax docs?